dump/src/util/dump/Dump.java

```diff
@@ -265,36 +265,39 @@ public Dump( Class<? extends E> beanClass, ObjectStreamProvider streamProvider,
       OPENED_DUMPS.add(this);
       _cacheSize = cacheSize;
       try {
-         checkVersion();
-
          if ( !isReadonly() && !_mode.contains(DumpAccessFlag.shared) ) {
             // the lock is released as soon as the RandomAccessFile is closed (stackoverflow.com/questions/421833)
             acquireFileLock();
          }
 
          readDeletions();
+         initMeta();
+
+         if ( !isReadonly() ) {
+            checkVersion();
```

Comment:
We should still check the version if read-only and throw if it mismatches.

Reply:
Ordinarily, we either rename, delete, or rewrite the dump on version mismatch. In the former cases, data has to be provided from another source again, whereas in the latter, we take existing contents through our normal schema update path, i.e. dropping the contents of removed fields.

This is sensible only if we intend to subsequently write to the dump, either from scratch or incrementally. Being read-only, this will not happen anyway. We might want to log any discrepancy, just like we do when we take care of it subsequently. Breaking things by throwing would be too much though, IMHO; at least in the Rewrite case.

I'll go about adding a log message in case version mismatches are found, and throw an exception whenever the dump would have been renamed or deleted. In case of Rewrite, we should be able to go about our business reading data from the existing dump.

Reply:
Now logging version mismatches, and throwing instead of renaming or deleting, while allowing entities annotated with Rewrite to go ahead.